lms/lmsweb/views.py

```diff
@@ -343,41 +363,51 @@ def upload_page():
     })
 
 
-@webapp.route(f'{routes.DOWNLOADS}/<int:solution_id>')
-@webapp.route(f'{routes.DOWNLOADS}/<int:solution_id>/<int:file_id>')
+@webapp.route(f'{routes.DOWNLOADS}/<string:endpoint>')
 @login_required
-def download(solution_id: int, file_id: Optional[int] = None):
-    solution = Solution.get_or_none(solution_id)
-    if solution is None:
+def download(endpoint: str):
```

Comment:
Change the name to "download_id" and add a documentation to explain it's a solution_id or share_id